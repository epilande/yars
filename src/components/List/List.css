.list {
  margin-top: 2rem;
  margin-bottom: 1.5rem;
  list-style: none;
}
