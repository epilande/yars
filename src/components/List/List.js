import React, { PropTypes } from 'react';
import ListItem from '../ListItem';
import styles from './List.css';

const List = ({ items, onRemove, onEdit, onComplete }) => (
  <ul className={styles.list}>
    {items.map(item =>
      <ListItem
        key={item.id}
        item={item}
        onRemove={onRemove}
        onEdit={onEdit}
        onComplete={onComplete}
      />
    )}
  </ul>
);

List.defaultProps = {
};

List.propTypes = {
  items: PropTypes.arrayOf(PropTypes.object),
  onRemove: PropTypes.func,
  onEdit: PropTypes.func,
  onComplete: PropTypes.func,
};

export default List;
