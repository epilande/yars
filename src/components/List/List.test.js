import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('<List />', () => {
  let component;
  const item = { id: 1, text: 'Hello' };

  beforeEach(() => {
    component = shallow(
      <List items={[item]} />,
    );
  });

  it('should render', () => {
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
