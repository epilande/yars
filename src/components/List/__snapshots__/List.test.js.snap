exports[`List should render a List 1`] = `
<ul
  className="list">
  <ListItem
    item={
      Object {
        "id": 1,
        "text": "Hello",
      }
    } />
</ul>
`;
